FSharpLu.Json/WithFunctor.fs

```diff
@@ -40,25 +44,56 @@ type With< ^S when ^S : (static member settings : JsonSerializerSettings)
         use jsonTextReader = new JsonTextReader(streamReader)
         serializer.Deserialize< ^T>(jsonTextReader)
 
+    /// Deserialize a stream to an object of type of targetType
+    static member inline public deserializeStreamToType targetType (stream:System.IO.Stream) =
+        let settings = (^S:(static member settings :  JsonSerializerSettings)())
+        let serializer = JsonSerializer.Create(settings)
+        use streamReader = new System.IO.StreamReader(stream)
+        use jsonTextReader = new JsonTextReader(streamReader)
+        serializer.Deserialize(jsonTextReader, targetType) :?> ^T
+
     /// Read Json from a file and desrialized it to an object of type ^T
     static member inline deserializeFile< ^T> file :^T =
         let settings = (^S:(static member settings :  JsonSerializerSettings)())
         System.IO.File.ReadAllText file |> With< ^S>.deserialize
+        
+    /// Read Json from a file and desrialized it to an object of targetType
+    static member inline deserializeFileToType targetType file =
+        let settings = (^S:(static member settings :  JsonSerializerSettings)())
```

Comment:
Can we get rid of those unused `settings` values?

Reply:
I don't think so, but you can try. This let binding is used to constrain the static type ^S.

Reply:
You are right: only the first 6 occurrences of `let settings = ..` in this module are needed, you can remove all the following ones where the `settings` binding is not actually used. @ChrSteinert